### PR TITLE
Fix pilot testflight distribution after recent iTC changes

### LINKED ISSUE #5309: pilot crashes with: submit_testflight_build_for_review!': [!] undefined method....
### Issue Description
This issue happens when running my beta lane: fastlane beta  

Everything runs soothly, until it tries to submit the app for review after the processing finishes with iTunes.

Pilot crashes the process with : submit_testflight_build_for_review!':[!] undefined method...

This happens in the pilot step in my fastfile to distribute to external:
```
pilot(distribute_external: true, "Some text here")
```

##### Complete output when running fastlane, including the stack trace and command used
```
/Library/Ruby/Gems/2.0.0/gems/spaceship-0.27.2/lib/spaceship/tunes/tunes_client.rb:618:in `submit_testflight_build_for_review!': [!] undefined method `[]' for nil:NilClass (NoMethodError)
	from /Library/Ruby/Gems/2.0.0/gems/pilot-1.9.2/lib/pilot/build_manager.rb:127:in `distribute_build'
	from /Library/Ruby/Gems/2.0.0/gems/pilot-1.9.2/lib/pilot/build_manager.rb:43:in `upload'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/actions/pilot.rb:18:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/runner.rb:197:in `block (2 levels) in execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/actions/actions_helper.rb:35:in `execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/runner.rb:182:in `block in execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/runner.rb:181:in `chdir'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/runner.rb:181:in `execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/runner.rb:107:in `trigger_action_by_name'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/fast_file.rb:140:in `method_missing'
	from Fastfile:87:in `block (2 levels) in parsing_binding'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/lane.rb:33:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/lane.rb:33:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/runner.rb:50:in `block in execute'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/runner.rb:46:in `chdir'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/runner.rb:46:in `execute'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/lane_manager.rb:46:in `cruise_lane'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/command_line_handler.rb:30:in `handle'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/commands_generator.rb:51:in `block (2 levels) in run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.48.1/lib/fastlane_core/ui/fastlane_runner.rb:26:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/commands_generator.rb:244:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/commands_generator.rb:20:in `start'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/lib/fastlane/cli_tools_distributor.rb:47:in `take_off'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.97.0/bin/fastlane:5:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
```

### Fastfile

Fastfile:
```ruby
desc "Submit a new Beta Build to Apple TestFlight"  
lane :beta do
  
    #default_payloads: [:test_result, :git_branch]
    #PULL GIT WITH LATEST CHANGES
    slack(
      message: "[FastLane] Starting BETA Integration.",
      success: true,        # optional, defaults to true   
      default_payloads: []
    )   
    
    ensure_git_branch(
      branch: 'develop'
      )
    
    git_pull                
    

  #SIGN PROVISIONING PROFILES      
  slack(
    message: "[FastLane] Signing Provisioning Profiles.",
    success: true,        # optional, defaults to true   
    default_payloads: []
  )   
  
  sigh

    #INCREMENT BULD NUMBER
    increment_build_number    
    build_number = Actions.lane_context[Actions::SharedValues::BUILD_NUMBER]
    # add_git_tag(tag: "testflight-#{build_number}")      

    #BUILD APP
    slack(
      message: "[FastLane] Compiling Build: #{build_number}",
      success: true,        # optional, defaults to true   
      default_payloads: []
    )   


    gym(scheme: "MyApp",
         clean: true        
      )       

    #COMMIT VERSION
    commit_version_bump(
      message: 'minor [CI] Build Bump'
      )    

      #PUSH GIT
      push_to_git_remote
      
      
    slack(
      message: "[FastLane] Build #{build_number} Now Commited. \n Delivering App. ",
      success: true,        # optional, defaults to true   
      default_payloads: []
    )   
      
    pilot(distribute_external: true, 
        changelog: "We constantly improve our app and deliver upgrades. This release includes bug fixes and performance enhancements."
    ) # Also Distribute to external testers
                           
    end
```

### Environment
fastlane version (run `fastlane -v`): fastlane 1.97.2

### PR DESCRIPTION
**Pull request by tmm1**
cc @tildedave

fixes #5309 #5394

### CODE REVIEW

File: spaceship/lib/spaceship/tunes/tunes_client.rb [hunk 1]
```diff
@@ -653,8 +645,9 @@ def submit_testflight_build_for_review!(app_id: nil, train: nil, build_number: n
     # rubocop:enable Metrics/ParameterLists
 
     def get_build_info_for_review(app_id: nil, train: nil, build_number: nil, platform: 'ios')
+      url = "ra/apps/#{app_id}/platforms/#{platform}/trains/#{train}/builds/#{build_number}/testInformation"
```
Comment (tildedave) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: nice.  this was a missing URL I couldn't figure out with my own hacking.

**Comment by KrauseFx** [comment]
This looks good to me, I'd just like a confirmation from one other user that this is working, then we can merge and push 🚀

**Comment by tildedave** [comment]
It looks like this needs some extra updates to handle the case where the export compliance data needs to be set, as previously the export compliance was added an posted as part of the `/submit/complete` route which isn't used any more.  I should have some shortly.

**Comment by tildedave** [comment]
@tmm1 https://github.com/tmm1/fastlane/pull/1 this worked for me.  

With the new setup of posting all the data as part of a request to `/review/submit`, we need to make sure that we are also providing export complicance/encryption data.

The big question I have here is around how we can determine whether or not to provide the export compliance data when submitting the build.

**Comment by tildedave** [comment]
I'll try to PR some test fixes sometime tomorrow 😄

**Comment by tmm1** [comment]
The tests are green now. Please merge and publish a new pilot gem. :boat:

**Comment by tmm1** [comment]
@KrauseFx @TKBurner ping

**Comment by mfurtak** [comment]
Looks good @tmm1 - thanks for this fix!

**Pull request merged**